spyder/plugins/ipythonconsole.py

```diff
@@ -1238,13 +1238,22 @@ def create_kernel_spec(self):
         ]
 
         # Environment variables that we need to pass to our sitecustomize
+        umr_namelist = CONF.get('main_interpreter', 'umr/namelist')
+
+        if PY2:
+            for umr_n in umr_namelist:
+                try:
+                    umr_n.encode('utf-8')
+                except UnicodeDecodeError:
+                    umr_namelist.remove(umr_n)
+            CONF.set('main_interpreter', 'umr/namelist', umr_namelist)

```

Comment:
I think we should be a bit smarter here and do this only if the result after removing the unicode modules is different from the original one.

So @mariacamilaremolinagutierrez, please make a copy of `umr_namelist` inside PY2 but before the `for` loop and apply CONF.set only if `umr_namelist != original_list`